### PR TITLE
Fix empty testdata file diff; fix simple diff bug

### PR DESCRIPTION
**Pull request by egibs**
Closes: https://github.com/chainguard-dev/bincapz/issues/349

This was tricky to track down. The file wasn't being skipped but the `fr.Behaviors` check was invalid because the slice was of length `0` rather than being `nil`:
```
fmt.Println(bool(fr.Behaviors == nil))
fmt.Println(bool(len(fr.Behaviors) == 0))
false
true
```

This PR fixes the bug and also addresses one small `Full` diff bug with the `simple` renderer; that is, `rep.Diff.Removed` was being referenced for added files rather than `rep.Diff.Added`.

### CODE REVIEW

File: samples/macOS/clean/ls.sdiff.level_2 [hunk 1]
```diff
@@ -1,2 +1,2 @@
-*** changed: macOS/clean/ls
+--- missing: .
```
Comment (egibs) [inline samples/macOS/clean/ls.sdiff.level_2:1]: Hmmm, I'll look into this as well.

File: pkg/render/markdown.go [hunk 2]
```diff
@@ -42,7 +42,9 @@ func matchFragmentLink(s string) string {
 }
 
 func (r Markdown) File(ctx context.Context, fr *bincapz.FileReport) error {
-	markdownTable(ctx, fr, r.w, tableConfig{Title: fmt.Sprintf("## %s [%s]", fr.Path, mdRisk(fr.RiskScore, fr.RiskLevel))})
+	if len(fr.Behaviors) != 0 {
```
Comment (tstromberg) [inline pkg/render/markdown.go:45]: NOTE: This further breaks `--omit-empty`, but we should get rid of it anyways, IMHO (see #348)

File: samples/macOS/clean/ls.sdiff.level_2 (hunk 1 above)
Comment (egibs) [reply samples/macOS/clean/ls.sdiff.level_2]: Resolved in [`0b94a09` (#350)](https://github.com/chainguard-dev/bincapz/pull/350/commits/0b94a097a4413b90e77b234b3560fe3c4d2917be).

File: samples/macOS/clean/ls.sdiff.trigger_2 [hunk 3]
```diff
@@ -1,5 +1,6 @@
-*** changed: macOS/clean/ls
```
Comment (tstromberg) [inline samples/macOS/clean/ls.sdiff.trigger_2:1]: I don't understand this output change either.

File: samples/macOS/clean/ls.sdiff.level_2 (hunk 1 above)
[the remark this answers is not in the document]
Comment (egibs) [reply samples/macOS/clean/ls.sdiff.level_2]: That was an unintended side-effect of my risk level check. I removed it and will instead rely on the `len(fr.Behaviors)...` check.

**Review by tstromberg — APPROVED** [review]

**Pull request merged**